src/filesystem/impls/filer/ArchiveUtils.js

```diff
@@ -85,13 +85,20 @@ define(function (require, exports, module) {
                 }
 
                 var isDir = file.options.dir;
+                var filename = removeProjectFolder(file.name);
                 filenames.push({
-                    absPath: Path.join(destination, file.name),
+                    absPath: Path.join(destination, filename),
                     isDirectory: isDir,
                     data: isDir ? null : new Buffer(file.asArrayBuffer())
                 });
             });
 
+            function removeProjectFolder(path){
+                // Nuke root folder thimble-project/ if exists
+                var regex = /^thimble\-project\//g;
```

Comment:
Let's combine these two lines:

```js
return path.replace(/^thimble\-project\//, "");
```

Also, I don't think you want to include the `g` in your regex, since you're only wanting to deal with the case of this string starting at the beginning.